### PR TITLE
allow Coverage to start when Coverage.running? is not implemented

### PR DESCRIPTION
**Pull request by notEthan**
Amends https://github.com/simplecov-ruby/simplecov/pull/1035/commits/1cc7650454fc7c53e7a724941662c3e9d3329ccc from #1035

Another option would be to just call Coverage.start unconditionally and rescue the RuntimeError it now raises when already started.

### CODE REVIEW

**Review by eregon — APPROVED** [review]
This looks good to me, thank you!

File: lib/simplecov.rb
Comment (eregon) [inline lib/simplecov.rb:361]: `Coverage.respond_to? :running?` seems nice than `rescue NoMethodError`, notably because it avoids an exception in case it's not available (and doesn't catch other issues potentially).

**Comment by eregon** [comment]
FYI I'm planning to try adding TruffleRuby in CI to catch such issues faster, but that needs more work so I would suggest merging this first.

**Comment by eregon** [comment]
Confirmed this PR reduces `bundle exec rake spec` from 9 to just 1 failure on truffleruby-dev.

File: lib/simplecov.rb
Comment (notEthan) [reply lib/simplecov.rb]: my thought was that the much more common case will have Coverage.running? implemented so checking for that every time would be slower than rescuing in the rarer case. but this is not a place where slowness matters at all, and I do prefer not to use exceptions for control flow, so agree with you and have updated it.

**Comment by notEthan** [comment]
I also started looking yesterday at adding truffle to the CI here, but got hung up on aruba/childprocess, then saw that you were ahead of me and had opened an issue about the same already.

**Comment by eregon** [comment]
@PragTob @colszowka Hello, it would be great if this can be merged and released when you have some open-source time, because the latest release breaks a bunch of CIs testing TruffleRuby (e.g. see backlinks to this PR)

**Comment by notEthan** [comment]
Yes, it would be quite helpful if this fix could make it into a release.

**Comment by eregon** [comment]
Ping, please merge and release this. It's been one month and a half and it still breaks the CI of e.g. https://github.com/ronin-rb/ronin/actions/runs/4089986844/jobs/7053088486
Can I do anything to help with this?

**Comment by AlexWayfer** [comment]
I've been stuck with this: https://github.com/filewatcher/filewatcher-matrix/pull/47

**Comment by PragTob** [comment]
Thanks for this, sorry for all the silence. Life and more happend, been a tough year+. I'll merge, it should be fine _famous simplecov maintainer last words_

Thanks y'all for your work :green_heart: 

@notEthan : Very cute bunny avatar!

**Comment by notEthan** [comment]
Thanks for merging. A release with this would make it usable.

**Comment by eregon** [comment]
@notEthan If the issue is SimpleCov 0.21.2 + TruffleRuby, that's solved in recent TruffleRuby releases and since https://github.com/simplecov-ruby/simplecov/issues/1044#issuecomment-1529783628, because TruffleRuby has `Coverage.running?` since then.

**Pull request merged**
